src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -43,6 +47,13 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     request.Path.ToUriComponent(),
                     request.QueryString.ToUriComponent());
 
+                // use a specific SSL port if it is provided in the options,
+                // clear it otherwise to use an empty default.
+                var uri = new UriBuilder(newUrl);
```

Comment:
I favored simplicity and efficiency over trying to re-invent port parsing and cause more issues. Was expecting this comment though and maybe someone knows a solution. If nobody chimes in I will keep it. And yes I confirm @davidfowl's comment is missing the fact that we need to clear the port when redirecting to https and `SslPort` is `null`, which fixes #4112.